### PR TITLE
Validator for password reset

### PR DESCRIPTION
**Pull request by msecret**
This includes code that can perform validations on regexes defined
through data attributes. It requires the user to write regexes on the
input element that can match up with the line items in a checkbox style
list.

This was done this way to ensure all configuration could be done easily
from within HTML, requiring less to be done in JS.

Refs #199

### CODE REVIEW

File: _components/form-blocks.md [hunk 1]
```diff
@@ -284,16 +284,21 @@ title: Forms Blocks
...
           <li>Another requirement</li>
         </ul>
       </div>
 
       <label for="password">New Password</label>
-      <input id="password" name="password" type="password">
+      <input id="password" name="password" type="password"
+        class="js-validate_password"
+        data-validate-length=".{8,}"
+        data-validate-uppercase="b"
```
Comment (juliaelman) [inline _components/form-blocks.md]: I think this regex validation got missed?

**Comment by juliaelman** [comment]
Looks good! There are some issues with the validation (see https://github.com/18F/govt-wide-patternlibrary/pull/264#discussion_r36466377) and movie below:

kiglnqzfw6

**Comment by msecret** [comment]
@juliaelman part of me feels like I want it to work like that, in that we don't want them to use a password that's only numbers. I read it as it should have at least 1 number, but shouldn't be only numbers.

**Comment by juliaelman** [comment]
@carodew @colinpmacarthur thoughts on https://github.com/18F/govt-wide-patternlibrary/pull/264#issuecomment-128516091 ?

**Comment by maya** [comment]
Wouldn't it not validate if the first one (have at least 1 uppercase character) wasn't met though? I think as soon as someone types a number that should check off. Otherwise there's no way to know and it seems confusing.

**Comment by carodew** [comment]
I agree with Maya and Julia. That specific requirement doesn't state that the password can't be all numbers, so I think we should stick with the literal interpretation. If there is a requirement for alpha characters that should be captured in a separate line.

**Comment by colinpmacarthur** [comment]
+1 to Carolyn

**Comment by juliaelman** [comment]
Cool - @msecret does that sound okay to you or do you need further clarification?

**Comment by juliaelman** [comment]
:+1: nice work!

**Pull request merged**